File: app/get_images.py
import urllib.request
import urllib.parse
import json

def search_unsplash(query):
    url = f"https://unsplash.com/napi/search/photos?query={urllib.parse.quote(query)}&per_page=1"
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode())
            if data['results']:
                return data['results'][0]['urls']['raw'].split('?')[0] + "?auto=format&fit=crop&q=80&w=800"
    except Exception as e:
        print(f"Error for {query}: {e}")
    return None

items = [
    "adjustable dumbbells",
    "dumbbells",
    "olympic barbell",
    "resistance bands",
    "treadmill",
    "yoga mat",
    "gym gloves",
    "protein shaker bottle",
    "weight plates gym",
    "kettlebell",
    "skipping rope",
    "battle rope gym",
    "wrist wraps gym",
    "weightlifting belt",
    "home gym equipment"
]

for item in items:
    url = search_unsplash(item)
    print(f"{item}|{url}")

File: app/applet/get_images.ts
import https from 'https';

const searchUnsplash = (query: string): Promise<string | null> => {
  return new Promise((resolve) => {
    const url = `https://unsplash.com/napi/search/photos?query=${encodeURIComponent(query)}&per_page=3&orientation=squarish`;
    https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
      let data = '';
      res.on('data', (chunk) => data += chunk);
      res.on('end', () => {
        try {
          const parsed = JSON.parse(data);
          if (parsed.results && parsed.results.length > 0) {
            const result = parsed.results[0];
            resolve(result.urls.raw.split('?')[0] + '?auto=format&fit=crop&q=80&w=800');
          } else {
            resolve(null);
          }
        } catch (e) {
          resolve(null);
        }
      });
    }).on('error', () => resolve(null));
  });
};

const items = [
  "adjustable dumbbells on rack",
  "heavy hex dumbbells gym",
  "olympic barbell gym",
  "fitness resistance bands",
  "running treadmill gym",
  "rolled yoga mat plain",
  "workout gym gloves",
  "protein shaker bottle gym",
  "weight plates barbell",
  "kettlebell gym floor",
  "digital skipping rope",
  "battle ropes gym",
  "gym wrist wraps",
  "leather weightlifting belt",
  "home gym equipment bundle"
];

async function run() {
  for (const item of items) {
    const url = await searchUnsplash(item);
    console.log(`${item} | ${url || 'NOT_FOUND'}`);
  }
}

run();
